test/RewardsPoolBase.js

```diff
@@ -0,0 +1,131 @@
+    let stakingTokenAddress;
+
+    let rewardTokensInstances;
+    let rewardTokensAddresses;
+	let rewardAmounts;
+	let rewardPerStakedToken
+
+
+    const rewardTokensCount = 1; // 5 rewards tokens for tests
+    const day = 60 * 24 * 60;
+	const amount = ethers.utils.parseEther("5184000");
+	const bOne = ethers.utils.parseEther("1");
+
+    beforeEach(async () => {
+		deployer = new etherlime.EtherlimeGanacheDeployer(aliceAccount.secretKey);
+		const currentBlock = await deployer.provider.getBlock('latest');
+		
+		stakingTokenInstance = await deployer.deploy(TestERC20, {}, amount);
+		await stakingTokenInstance.mint(aliceAccount.signer.address,amount);
+
+        stakingTokenAddress = stakingTokenInstance.contractAddress;
+        rewardTokensInstances = [];
+        rewardTokensAddresses = [];
+		rewardAmounts = [];
+		rewardPerStakedToken = [];
+        for (i = 0; i < rewardTokensCount; i++) {
+            let tknInst = await deployer.deploy(TestERC20, {}, amount);
+
+            // populate tokens
+            rewardTokensInstances.push(tknInst);
+			rewardTokensAddresses.push(tknInst.contractAddress);
+			rewardPerStakedToken.push(0)
+
+			// populate amounts
+			let rewardAmount = i + 1
+			let parsedReward = await ethers.utils.parseEther(rewardAmount.toString());
+            rewardAmounts.push(parsedReward);
+        }
+        RewardsPoolBaseInstance = await deployer.deploy(
+            RewardsPoolBase,
+            {},
+            stakingTokenAddress,
+            rewardAmounts,
+            rewardTokensAddresses,
+			(currentBlock.number + 40 ),
+			rewardPerStakedToken
+		);
+	});
+
+	it("Should deploy the RewardsPoolbase properly", async() => {
+		assert.isAddress(RewardsPoolBaseInstance.contractAddress, "The RewardsPoolBase contract was not deployed");
+
+        for (i = 0; i < rewardTokensInstances.length; i++) {
+            assert.isAddress(rewardTokensInstances[i].contractAddress, "The reward token contract was not deployed");
+        }
+	});
+
+	it("Should successfully stake", async() => {
+		const standardStakingAmount = ethers.utils.parseEther('10') // 10 tokens
+		await stakingTokenInstance.approve(RewardsPoolBaseInstance.contractAddress, standardStakingAmount);
+		for (i = 0; i < 50; i++) {
+			//forcing to mine some blocks
+			await stakingTokenInstance.mint(aliceAccount.signer.address,amount);
+        }
+		await RewardsPoolBaseInstance.stake(standardStakingAmount);
+		
+		let totalStakedAmount = await RewardsPoolBaseInstance.totalStakedAmount();
+
+		let userInfo = await RewardsPoolBaseInstance.userInfo(aliceAccount.signer.address)
+
+		assert(totalStakedAmount.eq(standardStakingAmount), "The stake was not successful")
+		assert(userInfo.amountStaked.eq(standardStakingAmount), "User's staked amount is not correct")
+		assert(userInfo.rewardDebt.eq(standardStakingAmount), "User's staked amount is not correct")
+	})
+
+	it("Should successfully stake and update the rates", async() => {
```

Comment:
Probably the most important test is to successfully stake and see that you accumulate a reward.